magenta/models/music_vae/js/index.ts

```diff
@@ -56,74 +49,73 @@ export class Nade {
 
   sample(encBias: dl.Tensor2D, decBias: dl.Tensor2D) {
     const batchSize = encBias.shape[0];
+    return dl.tidy(()=> {
+      let samples: dl.Tensor1D[] = [];
+      let a = encBias.clone();
+
+      for (let i = 0; i < this.numDims; i++) {
+        let h = dl.sigmoid(a);
+        let encWeights_i = dl.slice2d(
+          this.encWeights, [i, 0], [1, this.numHidden]).as1D();
+        let decWeightsT_i = dl.slice2d(
```

Comment:
I recommend aggressively using const :)

BTW - you might want to use a linter with travis (or VSCode settings) to automate some of this.

Checkout our tslint: https://github.com/PAIR-code/deeplearnjs/blob/master/tslint.json

And our .vscode settings we check in: https://github.com/PAIR-code/deeplearnjs/tree/master/.vscode

Reply:
Done.